src/ga/ga_selectors.rs

```diff
@@ -136,55 +128,46 @@ impl<'a, T: GASolution> GASelector<'a, T> for GARankSelector<'a, T>
         }
 
         // Select any individual from those that share best score.
-        self.population.individual(ga_random::ga_random_range(0, best_count), population_sort_basis)
+        population.individual(ga_random::ga_random_range(0, best_count), population_sort_basis)
     }
 }
 
-pub struct GAUniformSelector<'a, T: 'a + GASolution>
+pub struct GAUniformSelector
 {
-    population: &'a mut GAPopulation<T>
 }
 
-impl<'a, T: GASolution> GAUniformSelector<'a, T>
+impl GAUniformSelector
 {
-    pub fn new(p: &'a mut GAPopulation<T>) -> GAUniformSelector<'a, T>
+    pub fn new() -> GAUniformSelector
     {
         GAUniformSelector
         {
-            population: p
         }
     }
 }
 
-impl<'a, T: GASolution> GASelector<'a, T> for GAUniformSelector<'a, T>
+impl<'a, T: GASolution> GASelector<'a, T> for GAUniformSelector
 {
-    fn assign(&mut self, population: &'a mut GAPopulation<T>)
-    {
-        self.population = population;
-    }
-
-    fn update(&mut self)
+    fn update(&mut self, population : &mut GAPopulation<T>)
     {
         // Need to sort first, because GAPopulation.individual() draws individuals
         // from the sorted lists.
-        self.population.sort();
+        population.sort();
```

Comment:
There are a few options, I think:
1.- population.sort() can be made idempotent, so that calling it a bunch of times doesn't matter, something in the lines of 
` 
sort(self) 
 {
    if (self.sorted)
      return;
   self.actually_sort()
   self.sorted = true
}
`
The only problem is now keeping 'self.sorted', but the API is pretty thin so it might be worth.

2.- We can change 'update' to be something like 'prepare_selector(population)' so that selectors can put the population in any special state they need before calling select() on them. A similar situation of having the selector have a 'ready' flag that we need to clear arises.